### PR TITLE
SOLR-15470: Convert v2 GET /schema APIs to annotations

### PR DESCRIPTION
**Pull request by gerlowskija**
# Description

Solr's been in the slow process of moving its v2 APIs away from the
existing apispec/mapping framework towards one that relies on more
explicit annotations to specify API properties.

# Solution

This commit converts the APIs from the core.SchemaRead,
core.SchemaRead.fields, core.SchemaRead.copyfields, and
core.SchemaRead.dynamicFields_fieldTypes apispec files to the new
framework.

# Tests

Mostly manual testing.  With many of the "convert stuff to the annotation framework" PRs I've been writing, I've been able to include tests that assert that the v2->v1 API mapping logic is correct.  (e.g. V2CollectionAPIMappingTest).  But the v2 schema-read APIs are virtually identical to their v1 counterparts, so there is no "API mapping logic" to even assert on as a part of this test.

Beyond manual testing, we just need to rely here on any existing v2 schema API tests continuing to pass.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/handler/admin/api/SchemaKeyAPI.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import static org.apache.solr.client.solrj.SolrRequest.METHOD.GET;
+
+/**
+ * V2 API for getting the name of the unique-key field for an in-use schema.
+ *
+ * This API (GET /v2/collections/collectionName/schema/uniquekey) is analogous to the v1
+ * /solr/collectionName/schema/uniquekey API.
+ */
+public class SchemaKeyAPI {
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/handler/admin/api/SchemaKeyAPI.java:34]: Worth saying `SchemaUniqueKeyAPI` ???

File: solr/core/src/java/org/apache/solr/handler/admin/api/SchemaZkVersionAPI.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import static org.apache.solr.client.solrj.SolrRequest.METHOD.GET;
+
+/**
+ * V2 API for checking the ZK version of an in-use schema.
+ *
+ * This API (GET /v2/collections/collectionName/schema/zkversion) is analogous to the v1
+ * /solr/collectionName/schema/zkversion API.
+ */
+public class SchemaZkVersionAPI {
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/handler/admin/api/SchemaZkVersionAPI.java:34]: This isn't really a request for a change, just more like I wish the term `zk` didn't bleed into our public APIs...    Sigh.   The conversion looks good!

**Review by epugh — APPROVED** [review]
LGTM.

**Comment by gerlowskija** [comment]
Precommit and tests on this pass - I'll plan on merging tomorrow morning if there are no objections?

**Comment by epugh** [comment]
> Precommit and tests on this pass - I'll plan on merging tomorrow morning if there are no objections?

ship it!

**Pull request merged**